internal/plugin/manager/nvml.go

```diff
@@ -27,7 +27,7 @@ type nvmlmanager manager
 
 // GetPlugins returns the plugins associated with the NVML resources available on the node
 func (m *nvmlmanager) GetPlugins() ([]plugin.Interface, error) {
-	rms, err := rm.NewNVMLResourceManagers(m.infolib, m.nvmllib, m.devicelib, m.config)
+	rms, err := rm.NewNVMLResourceManagers(m.infolib, m.nvmllib, m.devicelib, m.config, m.imexChannels)
```

Comment:
Is this needed anymore since the list of imex channels is embedded in the config?

Reply:
This is still needed. The config values are used to construct a set of `imex.Channels`. The existence of device nodes is only checked at the point of construction.

Reply:
Is this just to avoid having to reevaluate all of this in each resource manager? It just feels very out of place alongside the other arguments to this function. Maybe I'm missing something though.

Reply:
Maybe we create a new type to pass in here called `rm.Settings`? Of which `ImexChannels` is the only element currently? That way we could at least extend it over time rather than continually adding more arguments to this function.

Reply:
I spent some time putting together https://github.com/NVIDIA/k8s-device-plugin/pull/991 which refactors the way in which we construct the plugin instances for each resource. I originally had that on top of this PR, but I think it can be reviewed separately from this one and I will rebase as required.